### DISCUSSION TITLE
[ideas] when `callr_function != NULL` and `debug != NULL`, `targets` should return error

### DISCUSSION THREAD

**Discussion by stemangiola** [ropensci/targets#1378] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I cannot see any use case where I would like to have `callr_function != NULL` and `debug != NULL`. 

When this happens accidentally (I forget to update the pipeline), the error due to side effects is not apparent, and this costs me long minutes to debug.

Should `targets` throw error if `callr_function != NULL` and `debug != NULL`?

**Comment by wlandau** [comment]
The case for `callr_function != NULL` and `debug != NULL` is not very common, but it could happen. You might have a generally error-prone target and may want to leave the `debug` option as is so you can easily return to it when `callr_function` is `NULL`. Otherwise, you might find yourself repeatedly commenting and uncommenting the `debug` option. So I would rather not go out of our way to throw an error in that case.

**Reply by stemangiola** [reply]
Thanks, just as a confirmation, if the debug is on, but the callr_function != NULL, does the debug get ignored, and is side effect free?

**Reply by wlandau** [reply]
Yes, under those conditions the debug option is ignored harmlessly.
